adr/2022-11-stable-spec.md

```diff
@@ -0,0 +1,178 @@
+
+The "Core Semantics" spec would contain the bare minimum rules that must be
+implemented for validators to not produce inaccurate results regardless of
+future revisions or extensions. Among other necessities, this would include a
+core set of keywords necessary to fully support structural validation and an
+extension mechanism. This spec should rarely change. New features would be added
+through additional specifications that define extensions to the "Core Semantics"
+spec.
+
+The "Standard Extensions" spec is an example of one of these extension
+specifications. This spec would be authored by the JSON Schema Org, but
+extension specifications could be authored by anyone. The "Standard Extensions"
+spec would include everything from the current spec that isn't included in the
+"Core Semantics" spec. Features and keywords included in this spec are so
+ubiquitous that they should be considered essential for implementations to
+support.
+
+### Option 3 - Minimal Change
+Option 3 represents the minimal amount of change to our process from what we
+have been doing. The spec would need to be converted from I-D XML to a Markdown
+version that would be served on the website, but otherwise we would continue to
+work the way we have been. We would aim for new version releases every year with
+patch releases mid-cycle. Each release is a distinct version of JSON Schema and
+has no compatibility guarantees between versions.
+
+## Decision Outcome
+The decision is to go with Option 1 while leaving discussion open for aspects of
+Option 2 that could be adopted within the constraints of Option 1.
+
+Option 2 uses an immutable spec where each release replaces the last while
+Option 1 uses a mutable spec. The outcome of having only one current version of
+the spec is achieved with either option, but the mutable spec allows us to
+remove some unnecessary roadblocks in our development processes and allows us to
+release a stable spec much sooner.
+
+Option 2's restructuring of the spec into "Core Semantics" and "Standard
+Extensions" isn't specifically ruled out, but spec evolution is expected to be
+done primarily through mutation of the spec guided by the stability process
+rather than through extension. Option 1 puts no constraint on the structure of
+the spec and restructuring is allowed at any time as long as it doesn't break
+compatibility requirements.
+
+## Pros and Cons of the Options
+The biggest benefit is shared between Option 1 and Option 2. Both approaches
+result in a stable spec. This will have benefits for both implementers and
+users. Because of the compatibility requirements, whenever you write a schema,
+you will never need to change it just to keep up with changes to JSON Schema.
+This is also better for implementers because they don't have to maintain
+separate code with different semantics in different versions. They just need to
+code for the current release and they will automatically have support for past
+releases.
```

Comment:
I think it's worth noting that "past releases" means "from when the first "stable" release is created, not including previous "draft era" versions of JSON Schema".
It's unlikely, but possible, that someone reading this might conclude that they should expect all previous versions, including "draf era" verisons of JSON Schema, should be supported in a single implementation which claims "stable era" support.